### ISSUE TITLE
Question regarding training flag.

### ISSUE THREAD

**Issue by m-dev12** [stanfordnlp/pyreft#139]

**Comment by m-dev12** [comment]
Hi @frankaging ,

If I print the {self.training flag} in the forward pass function of Loreftintervention, I see it to be false after the first epoch. 
Is this behavior expected? How else can I track if model is in train or eval mode? 

I was trying to figure out why this may be the case:
My hypothesis is that evaluate in class ReftTrainerForSequenceClassification(ReftTrainer) runs after each epoch. 

<img width="438" alt="image" src="https://github.com/user-attachments/assets/8b8ab339-2031-40a8-a821-9875cbbcf17b">

but the training step in HF trainer does not turn on the training mode in the same way?
 https://github.com/huggingface/transformers/blob/174890280b340b89c5bfa092f6b4fb0e2dc2d7fc/src/transformers/trainer.py#L3311

<img width="896" alt="image" src="https://github.com/user-attachments/assets/4b608c57-b25c-45a9-a2c8-24c03410caf6">

the loreft scripts also had this line https://github.com/stanfordnlp/pyreft/blob/bc8a49c6e5307e7d67c910292d4035a1384c1790/examples/loreft/original_code/task_steer.py#L261C5-L261C29 before training.

Could you help me understand if this is expected or this is an issue? How else can I track if model is in train or eval mode?

Thanks!

**Comment by m-dev12** [comment]
Also, if this is an issue it can have implications on whether interventions are being trained on in subsequent epochs, or is it just the classifier head being trained?

**Comment by frankaging** [comment]
@m-dev12 Hey, thanks for the input - it might take me a while to get back to this with actual testings since I am busy with other stuff right now. But here are a couple of pointers for you to test out stuff:

- set `allow_cls_grad` to `False` and see if you can still train? i think even with a random head, it will get pretty decent accuracy. This also means the interventions are receiving gradients and the optimizer is updated their weights.
- print out weights at each step?
- for all of other experiments, there is no head and the interventions are trainable, so probably it does not mean we are only training the head here?
- this issue could arise because of transformers versioning as well.

Keep us posted of your findings!

**Comment by m-dev12** [comment]
Sure @frankaging thanks for the input! Let me try out a few things next week, I'll let you know. And yes, version might also be an issue, since  I am on the latest transformer version.
